tests/MaxMind/Test/ReportTransactionTest.php

```diff
@@ -0,0 +1,320 @@
+<?php
+
+namespace MaxMind\Test;
+
+use MaxMind\Test\ReportTransactionData as Data;
+
+/**
+ * @coversNothing
+ */
+class ReportTransactionTest extends ServiceClientTest
+{
+    public function testMinimalRequest()
+    {
+        $this->assertEmpty(
+            $this->createReportTransactionRequest(
+                Data::minimalRequest(),
+                1
+            )->reportTransaction(Data::minimalRequest()),
+            'response for minimal request'
+        );
+    }
+
+    public function testFullRequest()
+    {
+        $req = Data::fullRequest();
+        $this->assertEmpty(
+            $this->createReportTransactionRequest(
+                $req
+            )->reportTransaction($req),
+            'response for full request'
+        );
+    }
+
+    public function testRequestsWithNulls()
+    {
+        $req = array_merge(
+            Data::minimalRequest(),
+            [
+                'chargeback_code' => null,
+                'maxmind_id' => null,
+                'minfraud_id' => null,
+                'notes' => null,
+                'transaction_id' => null,
+            ]
+        );
+        $this->assertEmpty(
+            $this->createReportTransactionRequest(
+                Data::minimalRequest(),
+                1
+            )->reportTransaction($req),
+            'response from request including nulls'
+        );
+    }
+
+    /**
+     * @expectedException \MaxMind\Exception\InvalidInputException
+     * @expectedExceptionMessage Must have keys
```

Comment:
I think these [annotations are deprecated](https://github.com/sebastianbergmann/phpunit/issues/3332) in new PHPUnit versions. We should be moving to the method form of testing. See [the docs](https://phpunit.readthedocs.io/en/9.0/writing-tests-for-phpunit.html#testing-exceptions).

Reply:
I've updated to the method form for `expectException`, `expectExceptionMessage` and `expectExceptionMessageRegExp`. Note that the last one [is deprecated now too](https://github.com/sebastianbergmann/phpunit/commit/d1199cb2e43a934b51521656be9748f63febe18e), and I tried to use it's successor `expectExceptionMessageMatches`, but it doesn't exist in the version of phpunit we've got listed in composer.json.